homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -0,0 +1,112 @@
+"""
+Interfaces with Egardia / Woonveilig alarm control panel.
+
+For more details about this platform, please refer to the $
+https://home-assistant.io/components/alarm_control_panel.egardia
+"""
+
+import logging
+
+import voluptuous as vol
+
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.components.alarm_control_panel import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_PORT, CONF_HOST, CONF_PASSWORD, CONF_USERNAME,
+    STATE_UNKNOWN, CONF_NAME,
+    STATE_ALARM_DISARMED, STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_AWAY)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pythonegardia>=1.0.3']
+REQUIREMENTS = ['requests']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Egardia'
+DEFAULT_PORT = '80'
+DOMAIN = 'egardia'
+NOTIFICATION_ID = 'egardia_notification'
+NOTIFICATION_TITLE = 'Egardia'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
+    })
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Egardia platform."""
+    name = config.get(CONF_NAME)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    add_devices([EgardiaAlarm(name, host, port, username, password)])
```

Comment:
In the setup you should check if the host is available and the password/username combination correct. If not, make the setup fail. Otherwise the users end-up with a non-functional platform in their setup.